android/app/src/main/java/app/organicmaps/bookmarks/BookmarksListFragment.java

```diff
@@ -603,6 +602,7 @@ public void onItemMore(int position)
   private void onDeleteTrackSelected(long trackId)
   {
     BookmarkManager.INSTANCE.deleteTrack(trackId);
+    mCategoryDataSource.invalidate();
```

Comment:
It will be an overhead for a programmer to find out that he should invalidate the dataset before performing operations. NotifyDatasetChanged is the easiest and most used way to perform updates in recycler view and it should handle all the updations keeping all these things abstracted as much as possible.